src/App.tsx

```diff
@@ -169,13 +200,49 @@ const App = () => {
 
 	const shadow = urlParams.get('shadow') !== 'false';
 
+	const form = { userCode: urlParams.get('user_code') || '' };
+
+	const width = getSizingValue({
+		urlParams,
+		key: 'width',
+		envVar: 'REACT_APP_FLOW_WIDTH'
+	});
+	const height = getSizingValue({
+		urlParams,
+		key: 'height',
+		envVar: 'REACT_APP_FLOW_HEIGHT'
+	});
+	const hasWidthHeight = width !== undefined || height !== undefined;
+
 	const containerClasses = clsx({
 		'descope-base-container': shadow,
-		'descope-wide-container': isWideContainer,
-		'descope-login-container': !isWideContainer
+		'descope-wide-container': !hasWidthHeight && isWideContainer,
+		'descope-login-container': !hasWidthHeight && !isWideContainer
 	});
 
-	const form = { userCode: urlParams.get('user_code') || '' };
+	// See: https://web.dev/blog/viewport-units
+	// This is sensitive to mobile
+	const flowContainerCss: CSSProperties = {
+		width: width !== undefined ? `min(${width}, 100dvw)` : undefined,
+		minHeight: height !== undefined ? `min(${height}, 100dvh)` : undefined
+	};
+
+	const bodyCss: CSSProperties = {};
+
+	try {
+		if (!background?.startsWith('https://')) {
+			// eslint-disable-next-line no-console
+			console.error(`background must be a https:// URL`);
+			throw new Error();
+		}
+		const url = new URL(background ?? '');
+		bodyCss.backgroundImage = `url(${url})`;
+		bodyCss.backgroundSize = 'cover';
```

Comment:
i suggest perhaps use `background` css that would allow customers to change the clip,size, and tons of other background options that i noticed are important for most scenarios. you can still keep `cover` by default

Reply:
Splitting into another PR (: